src/blocks/recommendations/class-recommendations-block.php

```diff
@@ -49,41 +49,38 @@ public function run(): void {
 	 *
 	 * @return void
 	 */
-	public static function register_block_and_assets(): void {
-		// Temporary workaround - don't register block when in FSE due to issues.
-		global $pagenow;
-		if ( 'site-editor.php' === $pagenow ) {
-			return;
-		}
+	public static function register_block(): void {
 
-		$plugin_url = plugin_dir_url( PARSELY_FILE );
+		// viewScript was added in WordPress 5.9 so we need to manually register our script if we're < 5.9.
+		// This can be removed if MINIMUM_WORDPRESS_VERSION is set to 5.9.
+		global $wp_version;
 
-		$editor_asset_file = require plugin_dir_path( PARSELY_FILE ) . 'build/recommendations-edit.asset.php';
-		wp_register_script(
-			'wp-parsely-recommendations-block-editor',
-			$plugin_url . 'build/recommendations-edit.js',
-			$editor_asset_file['dependencies'],
-			$editor_asset_file['version'],
-			true
-		);
+		if ( ! isset( $wp_version ) || version_compare( $wp_version, '5.9' ) < 0 ) {
+			$plugin_url      = plugin_dir_url( PARSELY_FILE );
+			$view_asset_path = plugin_dir_path( PARSELY_FILE ) . 'build/blocks/recommendations/view.asset.php';
+			if ( file_exists( $view_asset_path ) ) {
```

Comment:
Assuming that the build should always succeed (and we want a hard fail if it doesn't / hasn't been run locally), what does this condition here give us?

Reply:
It's an extra safeguard against a fatal error if we try to `require` a file that doesn't exist, but that's about it. Happy to remove this check if you feel it's superfluous.

Reply:
Under what conditions could we reasonably expect the file not to exist, and NOT want a fatal?

Reply:
This would only become an issue when/if the build directory was either non-existent or this file was missing. 

From a developer's point of view, it's not really a big deal to white screen the site if the build didn't get run, but in the (edge)-case of a plugin update going sideways on a client site, white screening the admin would be problematic.  At any rate, I have removed the check.